pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -195,66 +204,203 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
+
+  registryValueNamesFixedCase.forEach((key, index) => {
+    const originalName = registryValueNames[index];
+
+    if (key === '') {
+      unknownValueNames.push(originalName);
     } else {
-      regValue = nativeReg.queryValue(regKey, k);
-      if (typeof schemaVal === 'boolean') {
-        if (typeof regValue === 'number') {
-          regValue = regValue !== 0;
-        } else {
-          console.debug(`Deployment Profile expected boolean value for key ${ k }`);
-          regValue = false;
-        }
+      regValue = readRegistryValue(schemaObj[key], regKey, pathParts, originalName);
+      if (regValue !== null) {
+        newObject ??= {};
+        (newObject as Record<string, any>)[key] = regValue;
       }
     }
-    if (regValue !== null) {
-      newObject ??= {};
-      (newObject as Record<string, any>)[k] = regValue;
+  });
+  if (unknownValueNames.length > 0) {
+    console.error(`Unrecognized value names in registry at ${ pathParts.join('\\') }: [${ unknownValueNames.join(', ') }]`);
+  }
+
+  return newObject;
+}
+
+function fixProfileKeyCase(key: string, schemaKeys: string[]): string {
+  const idx = schemaKeys.findIndex(val => isEquivalentIgnoreCase(key, val));
+
+  return idx === -1 ? '' : schemaKeys[idx];
+}
+
+function readRegistryObject(regKey: nativeReg.HKEY, pathParts: string[], isSpecialObject = false) {
+  const newObject: Record<string, string[]|string|boolean|number> = {};
+
+  for (const k of nativeReg.enumValueNames(regKey)) {
+    let newValue = readRegistryValue(undefined, regKey, pathParts, k, isSpecialObject);
+
+    if (newValue !== null) {
+      if (isSpecialObject) {
+        switch (newValue) {
+        case 0:
+          newValue = false;
+          break;
+        case 1:
+          newValue = true;
+        }
+      }
+      newObject[k] = newValue;
     }
   }
 
   return newObject;
 }
 
+function readRegistryValue(schemaVal: any, regKey: nativeReg.HKEY, pathParts: string[], valueName: string, isSpecialObject = false): string[]|string|boolean|number|null {
+  const fullPath = `\\${ pathParts.join('\\') }\\${ valueName }`;
+  const valueTypeNames = [
+    'NONE', // 0
+    'SZ',
+    'EXPAND_SZ',
+    'BINARY',
+    'DWORD',
+    'DWORD_BIG_ENDIAN',
+    'LINK',
+    'MULTI_SZ',
+    'RESOURCE_LIST',
+    'FULL_RESOURCE_DESCRIPTOR',
+    'RESOURCE_REQUIREMENTS_LIST',
+    'QWORD',
+  ];
+  const rawValue = nativeReg.queryValueRaw(regKey, valueName);
+  const parsedValueForErrorMessage = nativeReg.queryValue(regKey, valueName);
+
+  if (rawValue === null) {
+    // This shouldn't happen
+    return null;
+  } else if (!isSpecialObject && schemaVal && typeof schemaVal === 'object' && !Array.isArray(schemaVal)) {
+    console.error(`Expecting registry entry ${ fullPath } to be a registry object, but it's a ${ valueTypeNames[rawValue.type] }, value: ${ parsedValueForErrorMessage }`);
+
+    return null;
+  }
+  const expectingArray = typeof schemaVal === 'object' && Array.isArray(schemaVal);
```

Comment:
```suggestion
  const expectingArray = Array.isArray(schemaVal);
```
The first part is unnecessary — any arrays must be an `object`.

Reply:
true